bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -0,0 +1,954 @@
+    parents::Parents,
+    payload::{
+        indexation::IndexationPayload,
+        milestone::{
+            MilestoneId, MilestonePayload, MilestonePayloadEssence, MILESTONE_MERKLE_PROOF_LENGTH,
+            MILESTONE_PUBLIC_KEY_LENGTH,
+        },
+        receipt::{MigratedFundsEntry, ReceiptPayload, TailTransactionHash, TAIL_TRANSACTION_HASH_LEN},
+        transaction::{Essence, RegularEssence, TransactionId, TransactionPayload},
+        treasury::TreasuryTransactionPayload,
+        Payload,
+    },
+    signature::{Ed25519Signature, SignatureUnlock},
+    unlock::{ReferenceUnlock, UnlockBlock, UnlockBlocks},
+    Message, MessageBuilder, MessageId,
+};
+use bee_protocol::types::peer::Peer;
+
+use serde::{Deserialize, Serialize, Serializer};
+use serde_json::Value;
+
+use std::convert::{TryFrom, TryInto};
+
+/// The message object that nodes gossip around in the network.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MessageDto {
+    #[serde(rename = "networkId")]
+    pub network_id: String,
+    #[serde(rename = "parentMessageIds")]
+    pub parents: Vec<String>,
+    pub payload: Option<PayloadDto>,
+    pub nonce: String,
+}
+
+impl From<&Message> for MessageDto {
+    fn from(value: &Message) -> Self {
+        MessageDto {
+            network_id: value.network_id().to_string(),
+            parents: value.parents().iter().map(|p| p.to_string()).collect(),
+            payload: value.payload().as_ref().map(Into::into),
+            nonce: value.nonce().to_string(),
+        }
+    }
+}
+
+impl TryFrom<&MessageDto> for Message {
+    type Error = Error;
+
+    fn try_from(value: &MessageDto) -> Result<Self, Self::Error> {
+        let mut builder = MessageBuilder::new()
+            .with_network_id(
+                value
+                    .network_id
+                    .parse::<u64>()
+                    .map_err(|_| Error::InvalidSyntaxField("networkId"))?,
+            )
+            .with_parents(Parents::new(
+                value
+                    .parents
+                    .iter()
+                    .map(|m| {
+                        m.parse::<MessageId>()
+                            .map_err(|_| Error::InvalidSyntaxField("parentMessageIds"))
+                    })
+                    .collect::<Result<Vec<MessageId>, Error>>()?,
+            )?)
+            .with_nonce_provider(
+                value
+                    .nonce
+                    .parse::<u64>()
+                    .map_err(|_| Error::InvalidSyntaxField("nonce"))?,
+                0f64,
+            );
+        if let Some(p) = value.payload.as_ref() {
+            builder = builder.with_payload(p.try_into()?);
+        }
+
+        Ok(builder.finish()?)
+    }
+}
+
+/// Describes all the different payload types.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum PayloadDto {
+    Transaction(Box<TransactionPayloadDto>),
```

Comment:
Are these just boxed because they're very big?

Reply:
The main reason was to have nested types. I used this to solve "recursion errors".
Also in bee-message we do use boxed payload types if i remember correctly.